### PR TITLE
Angular: Ivy rendering for Canvas and Docs

### PR DESCRIPTION
**Pull request by ThibaudAV**
Issue: 🤔 

## What I did

new version of this [PR](https://github.com/storybookjs/storybook/pull/15002) but without new "docs to render stories out of context" changes 

Sorry about refactoring. 😈
But to many case to handle like navigate between canvas and docs view or some particularities on each one need to use some patten to try to have clean code and not many function intertwined.
I hope it's better

- For Canvas:
Generate an instance of render by id which allows to find it for a partial rendering. (when only the props change the application is not re-rendered)
Reset the PlatformBrowserDynamic before each rendering, if is already exist 

- For Docs:
Performs a full rendering with each change (Same behavior but without props optimization)
Reset the PlatformBrowserDynamic only when the user leaves the docs page
Must be called sequentially

works with ivy and this `storybook-addon-angular-ivy` addon too

## How to test

- Is this testable with Jest or Chromatic screenshots? Yes
- Does this need a new example in the kitchen sink apps? No
- Does this need an update to the documentation? No

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

File: app/angular/src/client/preview/render.ts [hunk 1]
```diff
@@ -1,32 +1,41 @@
...
+  id,
 }: {
   storyFn: StoryFn<StoryFnAngularReturnType>;
   showMain: () => void;
   forceRender: boolean;
   parameters: Parameters;
+  targetDOMNode: HTMLElement;
+  id: string;
 }) {
-  showMain();
```
Comment (tmeasday) [inline app/angular/src/client/preview/render.ts:20]: This is what is tripping up Chromatic -- we need to call `showMain()` at some point to remove the `.sb-show-nopreview` class from the body.

Comment (shilman) [reply app/angular/src/client/preview/render.ts:20]: Nice catch!

Comment (ThibaudAV) [reply app/angular/src/client/preview/render.ts:20]: Nice, I must have lost it I don't know how

**Review by shilman — APPROVED** [review]
I'm in love, again. 😻

**Pull request merged**